### PR TITLE
Add a way to construct custom position in TASTy Reflect

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
@alemannosara here I will add a way for you to create the precise positions inside string literals you need.

**Comment by smarter** [comment]
I thought we discussed this and decided to not allow users to define custom positions since they're too easy to get wrong which might lead to crash when they're checked later ?

**Comment by nicolasstucki** [comment]
This PR doesn't actually allow the position to be set on a tree. The only way to use it is in an error message. Though I could add a special error method that receives custom offsets instead.

File: library/src/scala/tasty/reflect/Kernel.scala [hunk 1]
```diff
@@ -1067,6 +1067,12 @@ trait Kernel {
   /** Source code within the position */
   def Position_sourceCode(self: Position): String
 
+  /** Create a position that ranges from start to end (exclusive) in the given source
+   *  @param start index of the start of the range (0 <= start < end)
+   *  @param end index of the end of the range (start < end <= sizeOf(pos.sourceFile))
+   */
+  def Position_withOffset(self: Position)(start: Int, end: Int): Position
```
Comment (liufengyun) [inline library/src/scala/tasty/reflect/Kernel.scala]: Then I see this API, the first guess is that the offset is _relative_ to the given position `self`. But in the implementation, only the source file is used. It seems this is better to be defined as a constructor for `Position` which takes sourceFile directly.

Comment (nicolasstucki) [reply library/src/scala/tasty/reflect/Kernel.scala]: That would be terrible, each source position will try to reload the file.

Comment (nicolasstucki) [reply library/src/scala/tasty/reflect/Kernel.scala]: Maybe we should have the source file abstraction in TASTy reflect

**Comment by nicolasstucki** [comment]
Replaced by  #6158

**Pull request closed without merging**